src/Microsoft.Azure.WebJobs.Host/Config/ExtensionConfigContextExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Host.Config
+{
+    /// <summary>
+    /// Helper Extension methods for extension configuration. 
+    /// </summary>
+    public static class ExtensionConfigContextExtensions
+    {
+        /// <summary>
+        /// Register a set of rules to handle binding to a given attribute. 
+        /// </summary>
+        /// <typeparam name="TAttribute">The attribute to bind to</typeparam>
+        /// <param name="context"></param>
+        /// <param name="validator">a validator function to invoke on the attribute before runtime. </param>
+        /// <param name="bindingProviders">list of binding providers to handle this attribute. 
+        /// If none of these handle the attribute, an error is thrown at indexing. </param>
+        public static void RegisterBindingRules<TAttribute>(
```

Comment:
Also, I don't see any tests or usages of these new extension methods in this review, so it's hard to see usage context. I suggest either adding tests, or moving this into your other Queue PR and having one of the bindings use these extensions, so we can see the usage.